### PR TITLE
Bump minimum numpy version to 1.20

### LINKED ISSUE #6818: xarray 2022.6.0 doesn't work well with numpy 1.20
### What happened?

Fail to import xarray when xarray 2022.6.0 and numpy 1.20 are installed. It seems xarray 2022.6.0 is not compatible with numpy 1.20. I also tried numpy 1.19 and 1.21, they all work well with xarray 2022.6.0.

### What did you expect to happen?

xarray 2022.6.0 should work well with numpy 1.20.

### Minimal Complete Verifiable Example

```Python
$ conda create --name xarraytest xarray=2022.6.0 numpy=1.20
$ conda activate xarraytest
$ conda list
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2022.6.15            ha878542_0    conda-forge
importlib-metadata        4.11.4           py39hf3d152e_0    conda-forge
importlib_metadata        4.11.4               hd8ed1ab_0    conda-forge
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
libblas                   3.9.0           15_linux64_openblas    conda-forge
libcblas                  3.9.0           15_linux64_openblas    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 12.1.0              h8d9b700_16    conda-forge
libgfortran-ng            12.1.0              h69a702a_16    conda-forge
libgfortran5              12.1.0              hdcd56e2_16    conda-forge
libgomp                   12.1.0              h8d9b700_16    conda-forge
liblapack                 3.9.0           15_linux64_openblas    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libopenblas               0.3.20          pthreads_h78a6416_0    conda-forge
libstdcxx-ng              12.1.0              ha89aaad_16    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libzlib                   1.2.12               h166bdaf_2    conda-forge
ncurses                   6.3                  h27087fc_1    conda-forge
numpy                     1.20.3           py39hd249d9e_2    conda-forge
openssl                   3.0.5                h166bdaf_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.4.3            py39h1832856_0    conda-forge
pip                       22.2               pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
python                    3.9.13          h2660328_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.9                      2_cp39    conda-forge
pytz                      2022.1             pyhd8ed1ab_0    conda-forge
readline                  8.1.2                h0f457ee_0    conda-forge
setuptools                63.2.0           py39hf3d152e_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.39.2               h4ff8645_0    conda-forge
tk                        8.6.12               h27826a3_0    conda-forge
typing_extensions         4.3.0              pyha770c72_0    conda-forge
tzdata                    2022a                h191b570_0    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
xarray                    2022.6.0           pyhd8ed1ab_0    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
zipp                      3.8.0              pyhd8ed1ab_0    conda-forge
zlib                      1.2.12               h166bdaf_2    conda-forge

$ python
Python 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:58:50) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import xarray as xr
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/miniconda/envs/xarraytest/lib/python3.9/site-packages/xarray/__init__.py", line 1, in <module>
    from . import testing, tutorial
  File "/home/user/miniconda/envs/xarraytest/lib/python3.9/site-packages/xarray/testing.py", line 9, in <module>
    from xarray.core import duck_array_ops, formatting, utils
  File "/home/user/miniconda/envs/xarraytest/lib/python3.9/site-packages/xarray/core/duck_array_ops.py", line 26, in <module>
    from . import dask_array_compat, dask_array_ops, dtypes, npcompat, nputils
  File "/home/user/miniconda/envs/xarraytest/lib/python3.9/site-packages/xarray/core/npcompat.py", line 72, in <module>
    _SupportsDType[np.dtype],
  File "/home/user/miniconda/envs/xarraytest/lib/python3.9/typing.py", line 277, in inner
    return func(*args, **kwds)
  File "/home/user/miniconda/envs/xarraytest/lib/python3.9/typing.py", line 1004, in __class_getitem__
    _check_generic(cls, params, len(cls.__parameters__))
  File "/home/user/miniconda/envs/xarraytest/lib/python3.9/site-packages/typing_extensions.py", line 92, in _check_generic
    raise TypeError(f"{cls} is not a generic class")
TypeError: <class 'numpy.typing._dtype_like._SupportsDType'> is not a generic class
>>> 
```

### MVCE confirmation

- [X] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [X] Complete example — the example is self-contained, including all data and the text of any traceback.
- [X] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [X] New issue — a search of GitHub Issues suggests this is not a duplicate.

### Relevant log output

_No response_

### Anything else we need to know?

_No response_

### Environment

<details>

</details>

### PR DESCRIPTION
**Pull request by headtr1ck**
- [x] Closes #6818
- [ ] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [x] ~~New functions/methods are listed in `api.rst`~~

Alternative to https://github.com/pydata/xarray/pull/6821

### CODE REVIEW

**Comment by dcherian** [comment]
Hmm... doesn't seem like we can do this yet:

```
Package           Required             Policy               Status
----------------- -------------------- -------------------- ------
python            3.8     (2019-10-14) 3.8     (2019-10-14) =
boto3             1.13    (2020-04-30) 1.18    (2021-07-15) <
bottleneck        1.3     (2019-11-13) 1.3     (2019-11-13) =
cartopy           0.19    (2021-04-23) 0.19    (2021-04-23) =
cdms2             3.1     (2018-11-02) 3.1     (2018-11-02) =
cfgrib            0.9     (2019-02-25) 0.9     (2019-02-25) =
cftime            1.4     (2021-02-01) 1.5     (2021-05-20) <
dask-core         2021.4  (2021-04-02) 2021.7  (2021-07-09) <
distributed       2021.4  (2021-04-02) 2021.7  (2021-07-10) <
flox              0.5     (2022-05-03) 0.5     (2022-05-03) =
h5netcdf          0.11    (2021-04-20) 0.11    (2021-04-20) =
h5py              3.1     (2020-11-07) 3.3     (2021-06-22) <
hdf5              1.10    (2017-09-18) 1.12    (2020-08-03) <
iris              2.4     (2020-02-21) 3.0     (2021-01-25) <
lxml              4.6     (2020-10-17) 4.6     (2020-10-17) =
matplotlib-base   3.4     (2021-03-31) 3.4     (2021-03-31) =
nc-time-axis      1.2     (2019-01-25) 1.3     (2021-06-11) <
netcdf4           1.5.3   (2019-03-24) 1.5     (2019-03-24) = (w)
numba             0.54    (2021-08-29) 0.53    (2021-03-12) > (!)
numpy             1.20    (2021-01-31) 1.19    (2020-07-16) > (!)
packaging         20.0    (2020-01-06) 21.0    (2021-07-03) <
pandas            1.2     (2020-12-26) 1.3     (2021-07-02) <
pint              0.17    (2021-03-22) 0.17    (2021-03-22) =
pseudonetcdf      3.1     (2019-10-18) 3.1     (2019-10-18) =
pydap             3.2     (2018-10-16) 3.2     (2018-10-16) =
rasterio          1.2     (2021-01-25) 1.2     (2021-01-25) =
scipy             1.6     (2020-12-31) 1.7     (2021-06-27) <
seaborn           0.11    (2020-09-08) 0.11    (2020-09-08) =
sparse            0.12    (2021-03-19) 0.12    (2021-03-19) =
toolz             0.11    (2020-09-24) 0.11    (2020-09-24) =
typing_extensions 3.7     (2019-02-25) 3.10    (2021-06-03) <
zarr              2.8     (2021-04-25) 2.8     (2021-04-25) =
```

**Comment by headtr1ck** [comment]
Ok seems @Illviljan and myself were a few month off xD https://github.com/pydata/xarray/pull/6821#issuecomment-1193378283.

Nvmd then :)

**Comment by Illviljan** [comment]
Oh no, I read the wrong table, sorry about that. :(

It's this table https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule :
`On Jan 31, 2023 drop support for NumPy 1.20 (initially released on Jan 31, 2021)`

**Comment by dcherian** [comment]
Looks like we can bump a bunch of others if you're up for it :)

**Comment by headtr1ck** [comment]
> It's this table https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule :
> `On Jan 31, 2023 drop support for NumPy 1.20 (initially released on Jan 31, 2021)`

We are dropping 1.19 here, where this table says:
"On Jun 21, 2022 drop support for NumPy 1.19"
What is now correct?

**Comment by keewis** [comment]
you can probably use the `minimum version CI` to check (we have a policy that slightly deviates from NEP29): `<` means we can bump and `>` means we have to downgrade (it will fail).

I think `1.19` should be fine.

**Comment by headtr1ck** [comment]
Ok I think I found my misunderstanding.

Instead of dropping versions older than 18 month we want the oldest supported version to be at least 18 month old.

**Comment by headtr1ck** [comment]
Anyone know why min all deps fails?
As far as I can tell the only requirement of netCDF is python 3.8?

**Comment by headtr1ck** [comment]
I got rid of the complete npcompat module.
Not sure if you actually want to keep it around in case new incompatibilities arise?
But I guess we can always add it back.

**Comment by headtr1ck** [comment]
@dcherian I see that you were working on this the last time.
Do you have any idea why mamba fails to resolve the environment (for min-all-deps)?
With netcdf4=1.5.3 it fails for some python dependencies and with 1.5.4 it fails to find the correct cmds2?

We should really merge this, since all CI runs are currently broken due to new flox requiring numpy>=1.20

**Comment by dcherian** [comment]
> Do you have any idea why mamba fails to resolve the environment (for min-all-deps)?
With netcdf4=1.5.3 it fails for some python dependencies and with 1.5.4 it fails to find the correct cmds2?

Time to call in the wizards. 

@ocefpaf something's wrong with the environment solution.
With `netcdf4 1.5.3`

```
- package netcdf4-1.5.3-mpi_mpich_py38h137e1d7_4 requires python_abi 3.8.* *_cp38, but none of the providers can be installed
```

With `netcdf4 1.5.4`
```
Encountered problems while solving:
    - package cdms2-3.1.0-py27h6091dcd_0 requires python >=2.7,<2.8.0a0, but none of the providers can be installed
```
@tomvothecoder, I see you're committing to cdms. Is this something you've run in to.

> since all CI runs are currently broken due to new flox requiring numpy>=1.20

sorry! trying to fix this in https://github.com/conda-forge/flox-feedstock/pull/22

**Comment by headtr1ck** [comment]
> sorry! trying to fix this in conda-forge/flox-feedstock#22

Thanks!
Do we have to clear some micromamba cache for this to work immediately or should it just work after some days?

**Comment by keewis** [comment]
The cache will be cleared every day, or on changes to the contents of the environment file. I just reran the workflow which reported a cache miss so let's see if the new environment works?

File: ci/requirements/min-all-deps.yml [hunk 1]
```diff
@@ -3,53 +3,51 @@ channels:
...
   # When upgrading python, numpy, or pandas, must also change
-  # doc/installing.rst and setup.py.
+  # doc/getting-started-guide/installing.rst and setup.cfg.
   - python=3.8
-  - boto3=1.13
+  - boto3=1.18
   - bottleneck=1.3
-  - cartopy=0.19
+  - cartopy=0.20
   - cdms2=3.1
```
Comment (dcherian) [inline ci/requirements/min-all-deps.yml:14]:
```suggestion
 #  - cdms2=3.1
```

File: ci/requirements/min-all-deps.yml [hunk 2]
```diff
@@ -3,53 +3,51 @@ channels:
...
-  # doc/installing.rst and setup.py.
+  # doc/getting-started-guide/installing.rst and setup.cfg.
   - python=3.8
-  - boto3=1.13
+  - boto3=1.18
   - bottleneck=1.3
-  - cartopy=0.19
-  - cdms2=3.1
+  - cartopy=0.20
+ #  - cdms2=3.1
```
Comment (dcherian) [inline ci/requirements/min-all-deps.yml:14]:
```suggestion
  #  - cdms2=3.1
```

**Comment by tomvothecoder** [comment]
> With `netcdf4 1.5.4`
> 
> ```
> Encountered problems while solving:
>     - package cdms2-3.1.0-py27h6091dcd_0 requires python >=2.7,<2.8.0a0, but none of the providers can be installed
> ```
> @tomvothecoder, I see you're committing to cdms. Is this something you've run in to?
> 
> > since all CI runs are currently broken due to new flox requiring numpy>=1.20
> 
> sorry! trying to fix this in [conda-forge/flox-feedstock#22](https://github.com/conda-forge/flox-feedstock/pull/22)

I haven't ran into this issue since I've only committed documentation and README updates for the repo.

@jasonb5 maintains the repo dependencies, so let's see if he can help. Jason, can you try taking a look at this issue? It looks like `cdms2` is breaking the xarray builds.

File: doc/whats-new.rst [hunk 3]
```diff
@@ -26,6 +26,16 @@ New Features
 Breaking changes
 ~~~~~~~~~~~~~~~~
 
+- The minimum versions of some dependencies were changed:
+
+  =============== ====== ====
+  Package         Old    New
+  =============== ====== ====
+  numba           0.53   0.54
```
Comment (dcherian) [inline doc/whats-new.rst:34]: should update

File: xarray/core/npcompat.py [hunk 4]
```diff
@@ -1,247 +0,0 @@
-# Copyright (c) 2005-2011, NumPy Developers.
-# All rights reserved.
-
-# Redistribution and use in source and binary forms, with or without
```
Comment (dcherian) [inline xarray/core/npcompat.py:4]: What do you think of leaving an empty file with the copyright notice.

It'll create a lot of conflicts if there's an open PR touching this file (and its not uncommon to have to add some compat code here)

**Comment by headtr1ck** [comment]
Is there a reason why the minimum versions are not in the additional requirements in setup.cfg?

**Comment by dcherian** [comment]
> Is there a reason why the minimum versions are not in the additional requirements in setup.cfg?

See #5578

**Comment by jasonb5** [comment]
@dcherian @tomvothecoder  `cdms2-3.1.0` is old, the latest, final version is `3.1.5`. Updating the version should probably resolve any dependency issues.

**Comment by headtr1ck** [comment]
We usually only fix the minor version, in this case `cdms2=3.1`. But it seems to work now...

LGTM!

**Pull request merged**